src/main/java/logic/impl/commands/StaticCommandFactory.java

```diff
@@ -13,11 +13,10 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.function.BiConsumer;
+import java.util.function.Function;
 import java.util.function.Predicate;
-import java.util.function.Supplier;
 
-public class StaticCommandFactory implements Supplier<Map<String,
-        Pair<Predicate<CommandFormat>, BiConsumer<CommandFormat, TaskDao>>>> {
+public class StaticCommandFactory implements Function<String, Pair<Predicate<CommandFormat>, BiConsumer<CommandFormat, TaskDao>>> {
```

Comment:
Персонально, я, наверное, сделал бы отдельный интерфейс, если уж хочется использовать пакет java.util.function
```suggestion
interface ICommandFactory extends Function<String, Pair<Predicate<CommandFormat>, BiConsumer<CommandFormat, TaskDao>>> {...}
```
а уже потом его бы использовал как основной, вместо огромной конструкции, использующейся сейчас
```suggestion
public class StaticCommandFactory implements IStaticFactory {...}
```
Это значительно увеличит читаемость кода, текущую реализацию довольно тяжело разбирать.

Reply:
Спасибо! Исправлю.